cartridges/int_algolia/cartridge/scripts/algolia/steps/algoliaProductDeltaIndex.js

```diff
@@ -387,23 +387,31 @@ exports.process = function(cpObj, parameters, stepExecution) {
         if (productFilter.isInclude(product)) {
-
-            // Pre-fetch a partial model containing all non-localized attributes, to avoid re-fetching them for each locale
-            var baseModel = new AlgoliaLocalizedProduct({ product: product, locale: 'default', attributeList: nonLocalizedAttributes, fullRecordUpdate: fullRecordUpdate });
-            for (let l = 0; l < siteLocales.size(); l++) {
-                let locale = siteLocales[l];
-                let indexName = algoliaData.calculateIndexName('products', locale);
-                let localizedProduct = new AlgoliaLocalizedProduct({ product: product, locale: locale, attributeList: attributesToSend, baseModel: baseModel, fullRecordUpdate: fullRecordUpdate });
-                algoliaOperations.push(new jobHelper.AlgoliaOperation(baseIndexingOperation, localizedProduct, indexName));
+            if (productFilter.isIncludeOutOfStock(product)) {
+                var baseModel = new AlgoliaLocalizedProduct({ product: product, locale: 'default', attributeList: nonLocalizedAttributes, fullRecordUpdate: fullRecordUpdate });
+                for (let l = 0; l < siteLocales.size(); l++) {
+                    let locale = siteLocales[l];
+                    let indexName = algoliaData.calculateIndexName('products', locale);
+                    let localizedProduct = new AlgoliaLocalizedProduct({ product: product, locale: locale, attributeList: attributesToSend, baseModel: baseModel, fullRecordUpdate: fullRecordUpdate });
+                    algoliaOperations.push(new jobHelper.AlgoliaOperation(baseIndexingOperation, localizedProduct, indexName));
+                }
+                jobReport.processedItemsToSend++;
+            } else {
+                // => product is out-of-stock and IndexOutOfStock=false => must delete from Algolia
+                for (let l = 0; l < siteLocales.size(); l++) {
+                    var locale = siteLocales[l];
+                    var indexName = algoliaData.calculateIndexName('products', locale);
+                    algoliaOperations.push(new jobHelper.AlgoliaOperation(deleteIndexingOperation, { objectID: cpObj.productID }, indexName));
+                }
+                jobReport.processedItemsToSend++;
             }
-            jobReport.processedItemsToSend++;
         }
     } else {
         for (let l = 0; l < siteLocales.size(); l++) {
-            var locale = siteLocales[l];
-            var indexName = algoliaData.calculateIndexName('products', locale);
-            algoliaOperations.push(new jobHelper.AlgoliaOperation(deleteIndexingOperation, { objectID: cpObj.productID }, indexName));
+            let localeIndexName = algoliaData.calculateIndexName('products', siteLocales[l]);
+            algoliaOperations.push(new jobHelper.AlgoliaOperation(deleteIndexingOperation, { objectID: cpObj.productID }, localeIndexName));
```

Comment:
This looks ok, but this part of the code is only run when the job is in variant-level mode AND there are no attributes computed from the base product.

When there are attributes computed from the base product such as `colorVariations`, we don't pass here, we enter in the `if (paramRecordModel === MASTER_LEVEL || attributesComputedFromBaseProduct.length > 0) {` condition above.
So the logic needs to be updated above too.